### PR TITLE
feat: concise method syntax for object literals #2019

### PR DESCRIPTION
**Pull request by technohippy**
⯈feat: #2019 

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: tests/parser/object-literal-method.ts.fixture.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const obj = {
+  add: function(a: i32, b: i32): i32 {
```
Comment (romdotdog) [inline tests/parser/object-literal-method.ts.fixture.ts:2]: This should be identical to the source file.
https://github.com/AssemblyScript/assemblyscript/pull/1788#discussion_r607950828

The proper way would be to parse to a method declaration, [like here](https://ts-ast-viewer.com/#code/MYewdgzgLgBCBGArGBeGBvAUDGBDAJvgBS4BcMAlgMwBMANDPOdTQJTO0bY4wBOAplACuvMHhgBqRgG5uAX0xzpQA) and modify the compiler to support it.

Comment (technohippy) [reply tests/parser/object-literal-method.ts.fixture.ts:2]: Thanks, "AST Viewer" is really useful.

**Comment by MaxGraey** [comment]
Hi, thanks for PR! Could you add test cases for getter and setters as well?

**Comment by technohippy** [comment]
@MaxGraey Thank you for your comment. I've added tests for getter/setter.

**Pull request closed without merging**